### PR TITLE
chore(docs): Update programmatically-create-pages-from-data.md

### PR DESCRIPTION
**Pull request by ewalpole**
Changing from Promise/Resolve to Async.

## Description

Tried using the Promise/Resolve example in my code and it was throwing errors. Found your debugging async page and used the example there (very similar) and it worked. For consistency, moved that example to the other page as well. Was there a reason you were using Promise/Resolve in this space?

### CODE REVIEW

File: docs/docs/programmatically-create-pages-from-data.md [hunk 1]
```diff
@@ -21,35 +21,29 @@ access to the
...
-        createPage({
-          path: node.fields.slug,
-          component: path.resolve(`./src/templates/blog-post.js`),
-          context: {
-            slug: node.fields.slug,
-          },
-        })
+    }
+  `).then(res => {
+    res.data.allMarkdownRemark.edges.forEach(edge => {
```
Comment (DSchau) [inline docs/docs/programmatically-create-pages-from-data.md]: Perhaps move this after the await? It tends to clean up the code if we can move off of .then blocks!

e.g.

```
const { data } = await graphql(`
 # query
`)

data.allMarkdownRemark.edges.forEach(edge => {
```

Comment (ewalpole) [reply docs/docs/programmatically-create-pages-from-data.md]: @DSchau I made the changes and tested it, works for me. Added the highlighting as it was in the previous code sample.

File: docs/docs/programmatically-create-pages-from-data.md [hunk 2]
```diff
@@ -21,37 +21,29 @@ access to the
...
-          context: {
-            slug: node.fields.slug,
-          },
-        })
-      })
-      // highlight-end
-      resolve()
+```js:title=gatsby-node.js
+exports.createPages = async function({ actions, graphql }) {
+  // highlight-start
```
Comment (DSchau) [inline docs/docs/programmatically-create-pages-from-data.md]: This should be moved down 👇 (where it was previously!)

Comment (ewalpole) [reply docs/docs/programmatically-create-pages-from-data.md]: @DSchau Now? :D Sorry this is my first docs fix. Still figuring it out.

Comment (DSchau) [reply docs/docs/programmatically-create-pages-from-data.md]: Please don't apologize!! We appreciate you taking the time--and this looks great now.

Thank you!

**Pull request merged**